### ISSUE TITLE
cy.wait() fails to catch XHR response with 'Timed out retrying'

### ISSUE THREAD

**Issue by MarkCarbonell98** [cypress-io/cypress#6605]
### Current behavior:

Cypress does not catch some `cy.wait('@... ')` calls. It is not a time issue, as our servers answer in less than 2s and, and the old timeout used to be 30000ms with the same bug presenting itself.

You can see the routes that it generates in the following screenshots. Together with the failing parts of `cy.wait()`. I've used cypress since a very long time, and i've never experienced this kind of behavior in `cy.wait()` before

My `cy.router()` function is quite complex, but it is tested and works properly. It generates a set of statements starting with `cy.server()` and then concatenates them with `.route()`, where the name of the method, route, stubbed response and even status can be configured. 

![Screenshot from 2020-03-01 22-04-28](https://user-images.githubusercontent.com/32847477/75633877-3a40ae80-5c09-11ea-994c-adc228aa945a.png)
![Screenshot from 2020-03-01 22-04-04](https://user-images.githubusercontent.com/32847477/75633880-3c0a7200-5c09-11ea-8fb1-34aed292c427.png)
![Screenshot from 2020-03-01 22-03-49](https://user-images.githubusercontent.com/32847477/75633883-3d3b9f00-5c09-11ea-982d-4a77fc79130c.png)
![Screenshot from 2020-02-29 19-38-52](https://user-images.githubusercontent.com/32847477/75633886-3f056280-5c09-11ea-8813-f012b7f748d1.png)

### Desired behavior:

`cy.wait()` correctly waits for an XHR request and catches the response if there is one.

### Test code to reproduce

```js
/// <reference types='Cypress'/>

import { setObjectPropertiesToValue } from '../../support/utils/utils'
import { errorMessagesClass, emptyFieldError } from '../../support/testData'

describe('Vehicles', () => {
    const requiredVehicleFormSelectors = ["dialog-department", "dialog-manufacturer", "dialog-year-of-manufacture",  "model-name", "dialog-serial-num", "dialog-internal-num", "dialog-segment"]
    const unrequiredVehicleFormSelectors = ["upload-image", "dialog-notes", "dialog-status", "dialog-attachment", "dialog-mast-type", "dialog-driver-license", "dialog-cost-center", "dialog-operation-location", "dialog-load-capacity", "dialog-construction-height", "dialog-lift-height", "dialog-speed", "dialog-motor-type", "dialog-operating-hours"]

    const routes = [
        ['allGraphqlAuth', 'graphql', 'post'],
        ['allImages', 'rest', 'post'],
        ['oneImage', 'rest', 'patch'],
        ['allGraphql', 'graphql', 'post'],
        ['oneSockjsNode', 'graphql', 'get']
    ]

    before(() => {
        cy.reseedDatabaseWithSeeder('MockDataSeeder')
    })
    
    beforeEach(function() {
        
        cy.clearLocalStorage()
        cy.clearCookies()

        cy.router(routes, true)


        cy
            .automaticLogin()
            .visit('vehicles')
            .wait([
                '@postAllGraphqlAuth',
                '@postAllGraphqlAuth',
            ])

        cy
            .fixture("vehicles.json")
            .then(vehicles => {
                this.vehicle = vehicles[0];
            })
        
        cy
            .get('[data-cy=add-new-vehicle]')
            .should("be.visible")
            .click()
        
    })

    context('Creates a new Vehicle', function() {

        it('It can create a new vehicle with all fields filled', function() {
            
            const { vehicle } = this;

            cy
                .fillAddNewVehicleForm(vehicle)
                .wait([
                    '@postAllGraphqlAuth', '@postAllGraphqlAuth'
                ])
                
                // TODO complete this test
        })

        it("Can't create an empty vehicle", function() {

            cy
                .get("[data-cy=vehicle-dialog]")
                .should("be.visible")
                .within(() => {

                    cy
                        .get('[data-cy=save-dialog-button]')
                        .scrollIntoView()
                        .should("be.visible")
                        .click()
                        .wait([
                            '@postAllGraphqlAuth',
                        ])

                    cy
                        .get(errorMessagesClass)
                        .should("have.length.gte", 6)
                        .each(errorMessage => {
                            expect(errorMessage).to.contain(emptyFieldError)
                        })
                })
        })

        it('Can create a vehicle with all required fields filled', function() {

            cy.reseedDatabaseWithSeeder()

            const { vehicle } = this;
            
            const requiredVehicle = setObjectPropertiesToValue(vehicle, null, unrequiredVehicleFormSelectors)

            cy
                .fillAddNewVehicleForm(requiredVehicle)
                .wait([
                    '@postAllGraphqlAuth',
                ])

            cy
                .get('[data-cy=vehicle-dialog]')
                .should("not.be.visible")
        })

        it('Cannot create a new vehicle with unrequired fields ', function() {
            
            const { vehicle } = this;

            const unrequiredVehicle = setObjectPropertiesToValue(vehicle, null, requiredVehicleFormSelectors);

            cy
                .fillAddNewVehicleForm(unrequiredVehicle)
                .wait([
                    '@postAllGraphqlAuth',
                ])

            cy
                .get('[data-cy=vehicle-dialog]')
                .should("be.visible")
                .within(() => {

                    cy
                        .get(errorMessagesClass)
                        .should("have.length.gte", 6)
                        .each(errorMessage => {
                            expect(errorMessage).to.contain(emptyFieldError)
                        })
                })
        })
    })
})
```

This is the `fillAddNewVehicleForm` method

```js
Cypress.Commands.add("fillAddNewVehicleForm", (data) => {

    return cy
        .get('[data-cy=vehicle-dialog]')
        .should("be.visible")
        .within(() => {

            const keys = Object.keys(data)
            cy
                .wrap(keys)
                .each(key => {

                    const value = data[key];
                    cy.handleVehiclesData(key, value);
                })

            cy
                .get('[data-cy=save-dialog-button]')
                .scrollIntoView()
                .should("be.visible")
                .click()
        })
})
```

### Versions

OS: Ubuntu 19.10
Kernel: 5.3.0-29-generic
Cypress: 4.1.0
node: v12.13.1
Chrome: 80

**Comment by jennifer-shehane** [comment]
You mention:

>I've used cypress since a very long time, and i've never experienced this kind of behavior in cy.wait() before

Did this start happening in a recent upgrade to Cypress? Can you narrow it down to a specific version that this behavior started happening in?

Outside of this, we are aware of a bug with `cy.wait()` timing out when listening on multiple GraphQL endpoints. I suspect this is a duplicate of https://github.com/cypress-io/cypress/issues/5999 The example you have given is too complex and cannot be run ourselves to confirm though. Does this look like your issue?

**Comment by MarkCarbonell98** [comment]
@jennifer-shehane thank you for the fast answer!

I observed this behavior since cypress 3.6.1. I just upgraded yesterday to 4.1.0 and the behavior is still present. I've only observed it in GraphQL endpoints. There are other tests which use the same endpoint, where cy.wait() also sporadically fails. Failures in this particular test are constant in 90% of executions.  

If this tells you something, when I restart the docker container and run the tests on a fresh container `cy.wait()` does not fail. But after the second execution of the tests, it starts to fail constantly on the same XHR requests. 

And yes, this looks very similar to #5999 so fell free to close and reference it.

I hope you find this useful. I am really happy that you've answered my issue. I've read a lot of issues written by you and Brian since a about two years ago. And I have to tell you you're doing an amazing job with cypress! It's the best UI testing framework in the world! Keep up the good work!

Thanks again :)
